bulk_insert.go

```diff
@@ -32,7 +32,48 @@ func BulkInsert(db *gorm.DB, objects []interface{}, chunkSize int, excludeColumn
 	return nil
 }
 
+// BulkInsertWithAssigningIDs executes the query to insert multiple records at once.
+// it will scan 'returning id' to [returningId] after every insert.
+// it's necessary to set insert_option="returning id" in *gorm.DB
+//
+// [returningId] must be a *[]uint(for integer) or *[]string(for uuid)
+//
+// [objects] must be a slice of struct.
+//
+// [chunkSize] is a number of variables embedded in query. To prevent the error which occurs embedding a large number of variables at once
+// and exceeds the limit of prepared statement. Larger size normally leads to better performance, in most cases 2000 to 3000 is reasonable.
+//
+// [excludeColumns] is column names to exclude from insert.
+func BulkInsertWithAssigningIDs(db *gorm.DB, returningId interface{}, objects []interface{}, chunkSize int, excludeColumns ...string) error {
+	typ := reflect.TypeOf(returningId)
+	if typ.Kind() != reflect.Ptr || typ.Elem().Kind() != reflect.Slice {
+		return errors.New("returningId must be a slice ptr")
+	}
+
+	allIds := reflect.ValueOf(returningId).Elem()
```

Comment:
As far as I tested, the current one does not affect the value of `returningId`.

```suggestion
	allIDs := reflect.Indirect(reflect.ValueOf(returningId))
```

Reply:
good idea